src/util.ts

```diff
@@ -452,7 +463,65 @@ export function createMultiLinkFromLinks(links: string[]): string {
 		hashString += `&${key}=${value}`
 	}
 
-	return url.toString()
+	const shortenedLink = shortenMultilink(url.toString())
+
+	return shortenedLink
+}
+
+export function shortenMultilink(link: string): string {
+	const url = new URL(link)
+	const params = new URLSearchParams(url.search)
+
+	const i = params.get('i')
+	if (!i) {
+		throw new Error('Error shortening the multilink')
+	}
+
+	const numbers = i.split(',').map((num) => parseInt(num, 10))
+	let grouped = []
+	let start = numbers[0]
+	let count = 1
+
+	for (let i = 1; i <= numbers.length; i++) {
```

Comment:
Wow, damn! It's so short and lean now!!!!!!!!!

I didn't know that you can do `array[someBugIndex]` in js - in python you would get an index out of range error. Very cool!